### PR TITLE
feat(pie-docs): DSW-1482 update modal overview page

### PR DESCRIPTION
**Pull request by FayeCarter**
## Author Checklist (complete before requesting a review)
- [ ] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests
- [ ] If it is a `PIE Docs` change, I have reviewed the Docs site preview
- [ ] If it is a component change, I have reviewed the Storybook preview
- [ ] If there are visual test updates, I have reviewed them properly before approving

### Reviewer 1
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### Reviewer 2
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### CODE REVIEW

**Comment by xander-marjoram** [comment]
Should the `modifier-header-close` have a close button in it? It looks to me like a duplicate of `modifier-footer-double`.

File: apps/pie-docs/src/components/modal/overview/overview.md [hunk 1]
```diff
@@ -0,0 +1,424 @@
...
+    do: {
+        type: usageTypes.text,
+        items: [
+            "Use to inform users about a task can can contain information, require decisions or involve multiple tasks."
+        ]
+    },
+    dont: {
+        type: usageTypes.text,
+        items: [
+            "Modal’s interrupt a user’s flow by design, so while effective when used correctly; Modal’s should be used sparingly to limit disruption to the user."
```
Comment (xander-marjoram) [inline apps/pie-docs/src/components/modal/overview/overview.md:36]:
```suggestion
            "Modals interrupt a user’s flow by design, so while effective when used correctly; Modals should be used sparingly to limit disruption to the user."
```

File: apps/pie-docs/src/components/modal/overview/overview.md [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+
+{% list {
+    type: listTypes.ordered,
+    items: [
+        "**Back (optional)**: The chevron Icon button allows the user to return back a stage in a multi-step modal.",
+        "**Title**: Gives the users an overview of the content.",
+        "**Close**: The close Icon button will close the modal without submitting any data.",
+        "**Main content**: Open slot for any content required.",
+        "**CTAs (optional)**: Single or dual call to action Buttons to outline the user’s next options.",
+        "**Overlay**: Screen overlay that obscures the on-page content (1.Elements/1. Backgrounds/3. Page/Overlay)."
```
Comment (xander-marjoram) [inline apps/pie-docs/src/components/modal/overview/overview.md:60]: What does this bit in brackets mean? It's not immediately obvious to me

File: apps/pie-docs/src/components/modal/overview/overview.md [hunk 3]
```diff
@@ -0,0 +1,424 @@
...
+    alt: "Anatomy of a modal.",
+    shouldShowPadding: true
+} %}
+
+{% list {
+    type: listTypes.ordered,
+    items: [
+        "**Back (optional)**: The chevron Icon button allows the user to return back a stage in a multi-step modal.",
+        "**Title**: Gives the users an overview of the content.",
+        "**Close**: The close Icon button will close the modal without submitting any data.",
```
Comment (xander-marjoram) [inline apps/pie-docs/src/components/modal/overview/overview.md:57]:
```suggestion
        "**Close (optional)**: The close icon button will close the modal without submitting any data.",
```

File: apps/pie-docs/src/components/modal/overview/overview.md [hunk 4]
```diff
@@ -0,0 +1,424 @@
...
+
+{% contentPageImage {
+    src: "../../../assets/img/components/modal/overview/modifier-header-back.png",
+    alt: "A modal with a back icon button in the header.",
+    width: 600
+} %}
+
+#### Close
+
+Use this Header modifier when if the modal can't be dismissed.
```
Comment (xander-marjoram) [inline apps/pie-docs/src/components/modal/overview/overview.md:93]: I think it should be the opposite
```suggestion
Use this Header modifier when the modal can be dismissed.
```

File: apps/pie-docs/src/components/modal/overview/overview.md [hunk 5]
```diff
@@ -0,0 +1,424 @@
...
+    width: 600
+} %}
+
+### Footer
+
+You can replace the footer Buttons with any Button or Button pair defined within the Button guidance.
+
+#### Double
+
+A maximum of two Buttons are allowed within the Modal’s footer, with the primary Button right aligned and the lower emphasis Button on the left.
```
Comment (xander-marjoram) [inline apps/pie-docs/src/components/modal/overview/overview.md:107]:
```suggestion
A maximum of two Buttons are allowed within the Modal’s footer, with the primary Button right-aligned and the lower-emphasis Button on the left.
```

File: apps/pie-docs/src/components/modal/overview/overview.md [hunk 6]
```diff
@@ -0,0 +1,424 @@
...
+
+---
+
+## Narrow
+
+### Sizes
+
+#### Full screen
+
+The Wide large and optionally medium size, has the corresponding responsive option of fullscreen at narrow.
```
Comment (xander-marjoram) [inline apps/pie-docs/src/components/modal/overview/overview.md:180]: This sentence doesn't read very naturally to me. Is it trying to say that large modals are fullscreen at narrow viewports, and medium modals have the option of being fullscreen at narrow viewports?

File: apps/pie-docs/src/components/modal/overview/sizes.json [hunk 7]
```diff
@@ -0,0 +1,39 @@
+{
+  "headings": [
+    "Size",
+    "Breakpoint",
+    "Width and height"
```
Comment (jamieomaguire) [inline apps/pie-docs/src/components/modal/overview/sizes.json:5]: Could we have separate columns for width, height, max width and max height?

File: apps/pie-docs/src/components/modal/overview/overview.md [hunk 8]
```diff
@@ -0,0 +1,424 @@
...
+
+{% contentPageImage {
+  src: "../../../assets/img/components/modal/overview/layout-position-center.png",
+  alt: "A modal that is positioned in the middle of the screen.",
+  shouldShowPadding: true
+} %}
+
+#### Top
+
+The Modal is positioned at the center of the screen by default.
```
Comment (xander-marjoram) [inline apps/pie-docs/src/components/modal/overview/overview.md:254]:
```suggestion
The Modal is positioned at the top of the screen by default.
```

File: apps/pie-docs/src/components/modal/overview/overview.md [hunk 9]
```diff
@@ -0,0 +1,424 @@
...
+
+{% contentPageImage {
+  src: "../../../assets/img/components/modal/overview/layout-position-top.png",
+  alt: "A modal that is positioned in the middle of the screen.",
```
Comment (xander-marjoram) [inline apps/pie-docs/src/components/modal/overview/overview.md:258]:
```suggestion
  alt: "A modal that is positioned at the top of the screen.",
```

File: apps/pie-docs/src/components/modal/overview/overview.md [hunk 10]
```diff
@@ -0,0 +1,424 @@
...
+  width: 600
+} %}
+
+### Body content
+
+When the Modal’s content is longer than the available Modal height, then the body section should scroll vertically with the header, and a footer fixed in place. Modal content should never scroll horizontally; instead, use a larger size modal.
+
+#### Footer pinned = true
+
+If the actions are required to be sticky, they are pinned to the bottom of the Modal container whilst the user scrolls to indicate there is more content below until.
```
Comment (xander-marjoram) [inline apps/pie-docs/src/components/modal/overview/overview.md:282]:
```suggestion
If the actions are required to be sticky, they are pinned to the bottom of the Modal container whilst the user scrolls to indicate there is more content below.
```

File: apps/pie-docs/src/components/modal/overview/overview.md [hunk 11]
```diff
@@ -0,0 +1,424 @@
...
+
+{% contentPageImage {
+  src: "../../../assets/img/components/modal/overview/overflow-body-content-footer-pinned.png",
+  alt: "A modal with a pinned footer.",
+  width: 400
+} %}
+
+#### Footer pinned = false
+
+If the actions aren’t required to be sticky and placed at the bottom of the content, and the passive footer is pinned to the bottom of the Modal container whilst the user scrolls to indicate there is more content below.
```
Comment (xander-marjoram) [inline apps/pie-docs/src/components/modal/overview/overview.md:292]: I think this parapraph could do with rewording

File: apps/pie-docs/src/components/modal/overview/overview.md [hunk 12]
```diff
@@ -0,0 +1,424 @@
...
+
+{% contentPageImage {
+  src: "../../../assets/img/components/modal/overview/overflow-body-content-footer-not-pinned.png",
+  alt: "A modal with no pinned footer.",
```
Comment (xander-marjoram) [inline apps/pie-docs/src/components/modal/overview/overview.md:296]:
```suggestion
  alt: "A modal with an unpinned footer.",
```

File: apps/pie-docs/src/components/modal/overview/overview.md [hunk 13]
```diff
@@ -0,0 +1,424 @@
...
+
+### Close
+
+#### Dismissible
+
+If the Modal can be dismissed by the user, they should be able to dismiss it by clicking the close Icon Button or anywhere on the overlay. Doing this will close the Modal.
+
+{% contentPageImage {
+  src: "../../../assets/img/components/modal/overview/interaction-close-dismissible.png",
+  alt: "A modal with a dissmissible close button.",
```
Comment (xander-marjoram) [inline apps/pie-docs/src/components/modal/overview/overview.md:312]:
```suggestion
  alt: "A modal with a dismissible close button.",
```

File: apps/pie-docs/src/components/modal/overview/overview.md [hunk 14]
```diff
@@ -0,0 +1,424 @@
...
+  shouldShowPadding: true
+} %}
+
+#### Non-dismissible
+
+If the Modal is not dismissible, the user has to complete the flow in order to close the Modal.
+
+{% contentPageImage {
+  src: "../../../assets/img/components/modal/overview/interaction-close-non-dismissible.png",
+  alt: "A modal with a dissmissible close button.",
```
Comment (xander-marjoram) [inline apps/pie-docs/src/components/modal/overview/overview.md:322]:
```suggestion
  alt: "A non-dismissible modal without a close button.",
```

File: apps/pie-docs/src/components/modal/overview/overview.md [hunk 15]
```diff
@@ -0,0 +1,424 @@
...
+
+---
+
+## Examples
+
+Outlines the atomic level interactive elements for the component.
+
+### LTR examples
+
+Here are some examples of Buttons in left to right context:
```
Comment (xander-marjoram) [inline apps/pie-docs/src/components/modal/overview/overview.md:334]:
```suggestion
Here are some examples of Buttons in a left-to-right context:
```

File: apps/pie-docs/src/components/modal/overview/overview.md [hunk 16]
```diff
@@ -0,0 +1,424 @@
...
+
+{% contentPageImage {
+    src: "../../../assets/img/components/modal/overview/example-ltr-stampcards.png",
+    alt: "A left to right example of a modal with form fileds in the modal body.",
```
Comment (xander-marjoram) [inline apps/pie-docs/src/components/modal/overview/overview.md:338]:
```suggestion
    alt: "A left-to-right example of a modal with form fields in the modal body.",
```

File: apps/pie-docs/src/components/modal/overview/overview.md [hunk 17]
```diff
@@ -0,0 +1,424 @@
...
+
+{% contentPageImage {
+    src: "../../../assets/img/components/modal/overview/example-ltr-personal-information.png",
+    alt: "A left to right example of tabs and cards in the modal body.",
+    shouldShowPadding: true
+} %}
+
+### RTL examples
+
+Here are some examples of Buttons in right to left context:
```
Comment (xander-marjoram) [inline apps/pie-docs/src/components/modal/overview/overview.md:356]:
```suggestion
Here are some examples of modals in a right-to-left context:
```

File: apps/pie-docs/src/components/modal/overview/overview.md [hunk 18]
```diff
@@ -0,0 +1,424 @@
...
+
+{% contentPageImage {
+    src: "../../../assets/img/components/modal/overview/example-rtl-stampcards.png",
+    alt: "A right to left example of a modal with form fileds in the modal body.",
```
Comment (xander-marjoram) [inline apps/pie-docs/src/components/modal/overview/overview.md:360]:
```suggestion
    alt: "A right-to-left example of a modal with form fields in the modal body.",
```

File: apps/pie-docs/src/components/modal/overview/overview.md [hunk 19]
```diff
@@ -0,0 +1,424 @@
...
+    shouldShowPadding: true
+} %}
+
+{% contentPageImage {
+    src: "../../../assets/img/components/modal/overview/example-rtl-my-account.png",
+    alt: "A right to left example of a modal with form fileds in the modal body.",
```
Comment (xander-marjoram) [inline apps/pie-docs/src/components/modal/overview/overview.md:366]: This is the same alt text as the above image, is that correct?

File: apps/pie-docs/src/components/modal/overview/overview.md [hunk 20]
```diff
@@ -0,0 +1,424 @@
...
+    shouldShowPadding: true
+} %}
+
+{% contentPageImage {
+    src: "../../../assets/img/components/modal/overview/example-rtl-personal-information.png",
+    alt: "A right to left example of a modal with form fileds in the modal body.",
```
Comment (xander-marjoram) [inline apps/pie-docs/src/components/modal/overview/overview.md:372]: Same again

File: apps/pie-docs/src/components/modal/overview/overview.md [hunk 21]
```diff
@@ -0,0 +1,424 @@
...
+### Modifiers
+
+#### Footer
+
+As well as the standard Modal footer modifiers, at a narrow size you have the option to have full width Buttons that are stacked.
+
+{% contentLayout %}
+  {% contentPageImage {
+      src: "../../../assets/img/components/modal/overview/narrow-modifier-footer-dual-action.png",
+      alt: "A modal with a single primary button in its footer on a mobile screen.",
```
Comment (xander-marjoram) [inline apps/pie-docs/src/components/modal/overview/overview.md:207]: Alt text should say two buttons?

File: apps/pie-docs/src/components/modal/overview/overview.md [hunk 22]
```diff
@@ -0,0 +1,424 @@
...
+
+{% contentPageImage {
+    src: "../../../assets/img/components/modal/overview/modifier-footer-double.png",
+    alt: "A modal with a close icon button in the header.",
+    width: 600
+} %}
+
+#### Single
+
+You can use the Passive modifier for the footer in instances where the user doesn't need to take an action. This will remove the buttons at the bottom of the Modal.
```
Comment (xander-marjoram) [inline apps/pie-docs/src/components/modal/overview/overview.md:117]: Should this paragraph be swapped with the one for "Passive" below?

File: apps/pie-docs/src/components/modal/overview/sizes.json (hunk 7 above)
Comment (FayeCarter) [reply apps/pie-docs/src/components/modal/overview/sizes.json:5]: @nazanin

File: apps/pie-docs/src/components/modal/overview/overview.md (hunk 2 above)
Comment (FayeCarter) [reply apps/pie-docs/src/components/modal/overview/overview.md:60]: I'm not actually sure, @nazanin 
It's for this image
Screenshot 2023-11-24 at 11 56 12

File: apps/pie-docs/src/components/modal/overview/overview.md (hunk 6 above)
Comment (FayeCarter) [reply apps/pie-docs/src/components/modal/overview/overview.md:179]: @nazanin

File: apps/pie-docs/src/components/modal/overview/overview.md (hunk 21 above)
Comment (FayeCarter) [reply apps/pie-docs/src/components/modal/overview/overview.md:206]: thanks

Comment (FayeCarter) [reply apps/pie-docs/src/components/modal/overview/overview.md:206]: I'd put the images backward not the alt text

**Comment by FayeCarter** [comment]
> A couple of questions:
> 
> 1. Should the `modifier-header-close` have a close button in it? It looks to me like a duplicate of `modifier-footer-double`.
> 2. Where are the structure images used? I can't see them on the preview for this PR or [feat(pie-docs): DSW-1482 update modal code page #1006](https://github.com/justeattakeaway/pie/pull/1006)

Hey,
1. I think you're right so I've switched them. Just to double check with @nazanin 
this was modifier-footer-double
image
this was modifier-header-close
image

2. They're going to be used in the style page. I added all the images now just to make it easier but can remove them in the PR if you'd prefer?

File: apps/pie-docs/src/components/modal/overview/overview.md (hunk 6 above)
Comment (FayeCarter) [reply apps/pie-docs/src/components/modal/overview/overview.md:179]: For reference Naz, the 75% is 
'The Wide small and optionally medium size, has the corresponding responsive option of 75% width at narrow.'

File: apps/pie-docs/src/components/modal/overview/overview.md (hunk 11 above)
Comment (FayeCarter) [reply apps/pie-docs/src/components/modal/overview/overview.md:291]: Should it just be

'If the actions aren’t required to be sticky and are placed at the bottom of the content, the passive footer is pinned to the bottom of the Modal container whilst the user scrolls to indicate there is more content below.'?

File: apps/pie-docs/src/components/modal/overview/overview.md (hunk 2 above)
Comment (nazanin) [reply apps/pie-docs/src/components/modal/overview/overview.md:60]: @xander-marjoram @FayeCarter That's the address where the token exist on design, but I actually have no idea why that piece of information exist, there was a rush to get this up and no optimization has been done on the content of modal's doc, so any recommendations and rewordings are welcome, you can delete it now, I'll track to see why it's there and an alternative

File: apps/pie-docs/src/components/modal/overview/overview.md (hunk 6 above)
Comment (nazanin) [reply apps/pie-docs/src/components/modal/overview/overview.md:179]: We have 2 terms for describing screen sizes in component structure, wide for desktop screens and narrow for mobile and smaller ones, It's saying that a modal (large size and potentially medium) turns into a full screen modal on mobile screens, and smaller size modals will 75% with of the whole screen, I'm not the native to reword this :D

File: apps/pie-docs/src/components/modal/overview/sizes.json (hunk 7 above)
Comment (nazanin) [reply apps/pie-docs/src/components/modal/overview/sizes.json:5]: Yeah that's a good point, I'll take a look at this on Monday

File: apps/pie-docs/src/components/modal/overview/overview.md (hunk 11 above)
Comment (nazanin) [reply apps/pie-docs/src/components/modal/overview/overview.md:291]: "indicates the presence of additional content below"?

File: apps/pie-docs/src/components/modal/overview/overview.md (hunk 2 above)
Comment (FayeCarter) [reply apps/pie-docs/src/components/modal/overview/overview.md:60]: 👍  thanks

File: apps/pie-docs/src/components/modal/overview/overview.md (hunk 6 above)
Comment (FayeCarter) [reply apps/pie-docs/src/components/modal/overview/overview.md:179]: Thanks I'll try and updated :)

Comment (FayeCarter) [reply apps/pie-docs/src/components/modal/overview/overview.md:179]: I have gone for 
Large modals will expand to fill the entire page width at narrow viewports.
and
Medium and small modals will expand to 75% page width at narrow viewports.
What do you think?

**Comment by xander-marjoram** [comment]
@FayeCarter Happy for the images to stay if they're going to be used in a following PR!

Do you know why `structure-wide-empty-footer.png` and `structure-wide-no-footer.png` seem to be the same picture? They sound pretty similar so this might be intentional, but it might be worth reusing the image if they're identical

File: apps/pie-docs/src/assets/styles/components/_contentImage.scss [hunk 23]
```diff
@@ -60,6 +59,16 @@ $contentImage-height-l--mobile:           400px;
     }
 }
 
+.c-contentImage--hasBackdrop .c-contentImage-backdrop,
+.c-contentImage--hasPadding {
+    background-color: f.$color-container-subtle;
+}
+
+.c-contentImage--hasPadding {
```
Comment (jamieomaguire) [inline apps/pie-docs/src/assets/styles/components/_contentImage.scss:67]: is hasPadding the most descriptive class name if it also includes rounded borders?

File: apps/pie-docs/src/components/modal/overview/overview.md [hunk 24]
```diff
@@ -0,0 +1,421 @@
...
+
+---
+
+## Dos and Don’ts
+
+{% usage {
+    do: {
+        type: usageTypes.text,
+        items: [
+            "Use to inform users about a task can can contain information, require decisions or involve multiple tasks."
```
Comment (jamieomaguire) [inline apps/pie-docs/src/components/modal/overview/overview.md:30]:
```suggestion
            "Use to inform users about a task can contain information, require decisions or involve multiple tasks."
```

File: apps/pie-docs/src/components/modal/overview/overview.md [hunk 25]
```diff
@@ -0,0 +1,421 @@
...
+## Narrow
+
+### Sizes
+
+#### Full screen
+
+Large modals will expand to fill the entire page width at narrow viewports.
+{% contentPageImage {
+    src: "../../../assets/img/components/modal/overview/narrow-size-full-screen.png",
+    alt: "A full screen modal on a mobile screen.",
```
Comment (jamieomaguire) [inline apps/pie-docs/src/components/modal/overview/overview.md:181]:
```suggestion
    alt: "A fullscreen modal on a mobile screen.",
```

File: apps/pie-docs/src/components/modal/overview/overview.md [hunk 26]
```diff
@@ -0,0 +1,421 @@
...
+### Position
+
+{% notification {
+  type: "information",
+  message: "Only applicable to Modals that aren’t full screen."
+} %}
+
+#### Center
+
+The Modal is positioned at the center of the screen by default.
```
Comment (jamieomaguire) [inline apps/pie-docs/src/components/modal/overview/overview.md:241]: The casing for Modal and Button seems a little inconsistent on this page. I personally prefer lowercase but please can you ensure it's the same everywhere whichever you choose?

File: apps/pie-docs/src/assets/styles/components/_contentImage.scss (hunk 23 above)
Comment (FayeCarter) [reply apps/pie-docs/src/assets/styles/components/_contentImage.scss:67]: Any suggestions? Not really sure what else to name it. It's just this part.
`backdropHasPadding`? `gapBetweenImageAndBackdrop`?
Screenshot 2023-11-30 at 11 24 37

Comment (jamieomaguire) [reply apps/pie-docs/src/assets/styles/components/_contentImage.scss:67]: off the top of my head `--rounded` and `--paddedBackdrop` or something (its a bit hard to tell which part of that image you mean - is the gap the pale grey space surrounding the image?

Comment (jamieomaguire) [reply apps/pie-docs/src/assets/styles/components/_contentImage.scss:67]: alternatively, leave hasPadding but move radius to `--rounded`

**Review by raoufswe — APPROVED** [review]

File: apps/pie-docs/src/_11ty/shortcodes/componentDetailsTable.js [hunk 27]
```diff
@@ -59,15 +59,25 @@ const buildRow = (cells) => cells.map((cell) => {
...
     const { headings, rows } = JSON.parse(tableData);
+    const hasWidePadding = headings?.length <= 3;
 
     return `<div class="c-componentDetailsTable-backdrop">
-    <table class="c-componentDetailsTable">
-    ${headings ? `<tr>${headings.map((heading) => `
-        <th>${heading}</th>`).join('')}
+    <table class="c-componentDetailsTable ${hasWidePadding ? 'c-componentDetailsTable-hasWidePadding' : ''}">
+    ${headings ? `<tr>${headings.map((heading, index) => {
+        const isPriorityColumn = priority && priority === index;
```
Comment (raoufswe) [inline apps/pie-docs/src/_11ty/shortcodes/componentDetailsTable.js:70]: do we still need this change?
if not, we can clean it up in a follow up pr

**Review by jamieomaguire — APPROVED** [review]

**Review by xander-marjoram — APPROVED** [review]

File: apps/pie-docs/src/_11ty/shortcodes/componentDetailsTable.js (hunk 27 above)
Comment (FayeCarter) [reply apps/pie-docs/src/_11ty/shortcodes/componentDetailsTable.js:70]: I'll do it on the next one, thanks :)

**Pull request merged**